eeg_models/datasets/braininvaders.py

```diff
@@ -0,0 +1,141 @@
+        """return data for a single subject"""
+
+        file_path_list = self.data_path(subject)
+        sessions = {}
+        for file_path in file_path_list:
+
+            session_number = file_path.split(os.sep)[-2].replace("Session", "")
+            session_name = "session_" + session_number
+            if session_name not in sessions.keys():
+                sessions[session_name] = {}
+
+            run_number = file_path.split(os.sep)[-1]
+            run_number = run_number.split("_")[-1]
+            run_number = run_number.split(".gdf")[0]
+            run_name = "run_" + run_number
+
+            raw_original = mne.io.read_raw_edf(
+                file_path, montage="standard_1020", preload=True
+            )
+
+            sessions[session_name][run_name] = raw_original
+
+        return sessions
+
+    def data_path(
+        self,
+        subject: int,
+        path: Optional[Directory] = None,
+    ) -> Optional[Directory]:
+
+        if subject not in self.subject_list:
+            raise (ValueError("Invalid subject number"))
+
+        url = "{:s}subject{:d}.zip".format(
+            "https://zenodo.org/record/1494240/files/", subject
+        )
+        path_zip = dt_path(url, "BRAININVADERS")
+        path_folder = path_zip.strip("subject{:d}.zip".format(subject))
+
+        if not (os.path.isdir(path_folder + "subject{:d}".format(subject))):
+            print("unzip", path_zip)
+            zip_ref = zipfile.ZipFile(path_zip, "r")
+            zip_ref.extractall(path_folder)
+
+        meta_file = os.path.join("subject{:d}".format(subject), "meta.yml")
+        meta = read_yaml(path_folder + meta_file)
+        conditions = []
+        if self.adaptive:
+            conditions = conditions + ["adaptive"]
+        if self.non_adaptive:
+            conditions = conditions + ["non_adaptive"]
+        types = []
+        if self.training:
+            types = types + ["training"]
+        if self.online:
+            types = types + ["online"]
+        filenames = []
+        for run in meta["runs"]:
+            run_condition = run["experimental_condition"]
+            run_type = run["type"]
+            if (run_condition in conditions) and (run_type in types):
+                filenames = filenames + [run["filename"]]
+
+        subject_paths = []
+        for filename in filenames:
+            subject_paths = subject_paths + glob.glob(
+                os.path.join(
+                    path_folder, "subject{:d}".format(subject), "Session*", filename
+                )
+            )
+        return subject_paths
+
+    def __len__(self) -> int:
+        return len(self.subject_list)
+
+    def __getitem__(self, index: int) -> Dict[str, Any]:
+        sessions = self._get_single_subject_data(index)
+        eegs, markers = [], []
+        for _, run in sorted(sessions["session_1"].items()):
+            r_data = run.get_data()
+            eegs.append(r_data[:-1])
+            markers.append(r_data[-1])
```

Comment:
This data is suspected to be just data channel, not markers.

Markers should be extracted with mne.events_from_annotations